src/schema/Schema.ts

```diff
@@ -92,39 +92,27 @@ export class InlineSchema {
 	readonly propNames: string[];
 
 	private readonly argsSugarsByStarts: SugarsByStart[];
-	private readonly props: InlinePropDefinition[];
+	readonly argsSchemas: ReadonlyArray<InlinePropDefinition>;
```

Comment:
This is linked to my comment below (in `SchemaDefinition.ts`). Rather than expose the schema definition object directly, I'd rather expose methods that allow us to query the schema and get the type of value that is most useful internally. This is for two reasons:

- I would rather have only `Schema` be used internally, and not a mix of `Schema` and `SchemaDefinition`
- I don't find it super pretty to expose a `true | undefined` field. I know it's convenient in order to write the schema, but it's not pretty internally.

Therefore, I think this should stay `private`, and we should have a getter that exposes the information instead.

Reply:
If we added an `ArgSchema` class and renamed our methods a little, we could have a really nice interface to query the schema:

```typescript
schema.inline("link").arg(1).raw
```